src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,505 @@
 
 #include "sync/remote_mongo_collection.hpp"
 
+namespace realm {
+namespace app {
+
+static void handle_response(util::Optional<AppError> error,
+                            util::Optional<std::string> value,
+                            std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        return completion_block(value.value(), error);
+    }
+    
+    return completion_block("", error);
+}
+
+static void handle_count_response(util::Optional<AppError> error,
+                                  util::Optional<std::string> value,
+                                  std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        try {
+            auto json = nlohmann::json::parse(value.value());
+            auto count_string = json.at("$numberLong").get<std::string>();
+            return completion_block(std::stoi(count_string), error);
```

Comment:
stoi returns an `int`, so this'll truncate values. `stoll` gives a 64-bit int.

Reply:
changed to stoll